src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -248,17 +263,44 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @return true if a right curly should be alone on a line.
      */
     private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy, Details details) {
+        final int tokenLambda = details.lcurly.getParent().getType();
         return bracePolicy == RightCurlyOption.ALONE
+                && tokenLambda != TokenTypes.LAMBDA
                 && !isAloneOnLine(details)
                 && !isEmptyBody(details.lcurly)
-                || bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
+                || shouldBeAloneInLambda(bracePolicy, details)
+                || shouldBeAloneOrSingleLine(bracePolicy, details);
+    }
```

Comment:
I do not like code that write in chunks .... 
Lets change a strategy.
Please ignore CC violations and make code as it should be.
I will do code review even you have CC problems.
Ones logic will be written correctly, I will look how how to refactor to pass CC myself.